Automattic/jetpack
# SEO: add a 'noindex' checkbox for posts and pages.

Closes #26925
Follow-up to #27236

#### Changes proposed in this Pull Request:

* Adds a 'noindex' checkbox setting for posts and pages.
* Posts marked with this setting are excluded from being added to the Jetpack sitemap.

#### Other information:


#### Jetpack product discussion

Internal: pbNhbs-4eC-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

- After pulling down this branch, make sure SEO and Sitemaps features are turned on in: `/wp-admin/admin.php?page=jetpack#/traffic`
- Create a new test post or page and publish it _without_ using the new "SEO Indexing" checkbox from this PR.
- Examine the published post source on the frontend and make sure there is not any `<meta name='robots' content='noindex'...` or similar.
  - If there is, double check your "Search engine visibility" in Reading settings: `/wp-admin/options-reading.php`
- Now edit the same post and tick the "Hide this page from search engines" checkbox and save.
- Refresh the post on the frontend to examine that it has a `<meta name="robots" content="noindex" />` tag.
- Using a plugin like "WP Crontrol" or wp-cli commands, run the `jp_sitemap_cron_hook` event.
- Examine your Jetpack sitemap for the now non-indexed post, it should be absent from the freshly generated sitemap.
- Setting the Jetpack SEO options should work equally from both the Jetpack sidebar menu and the pre-publish panel.

**Jetpack menu:**

![jetpack-seo-panel](https://user-images.githubusercontent.com/15803018/203676207-99ee06b8-2136-414b-9271-24847ff27f10.png)

**Pre-publish panel:**

![jetpack-seo-pre-publish-panel](https://user-images.githubusercontent.com/15803018/203676224-124e7f66-b484-48dd-9ec3-00268029acc5.png)
